ui/src/app/base/components/StatusBar/StatusBar.test.tsx

```diff
@@ -15,119 +13,155 @@ import {
   rootState as rootStateFactory,
   versionState as versionStateFactory,
 } from "testing/factories";
-
-const mockStore = configureStore();
-
-describe("StatusBar", () => {
-  let state: RootState;
-
-  beforeEach(() => {
-    jest.useFakeTimers("modern");
-    // Thu, 31 Dec. 2020 23:00:00 UTC
-    jest.setSystemTime(new Date(Date.UTC(2020, 11, 31, 23, 0, 0)));
-    state = rootStateFactory({
-      config: configStateFactory({
-        items: [configFactory({ name: "maas_name", value: "bolla" })],
-      }),
-      general: generalStateFactory({
-        version: versionStateFactory({ data: "2.10.0" }),
-      }),
-      machine: machineStateFactory({
-        active: "abc123",
-        items: [],
-      }),
-    });
+import { renderWithMockStore } from "testing/utils";
+
+let state: RootState;
+beforeEach(() => {
+  jest.useFakeTimers("modern");
+  // Thu, 31 Dec. 2020 23:00:00 UTC
+  jest.setSystemTime(new Date(Date.UTC(2020, 11, 31, 23, 0, 0)));
+  state = rootStateFactory({
+    config: configStateFactory({
+      items: [configFactory({ name: "maas_name", value: "bolla" })],
+    }),
+    general: generalStateFactory({
+      version: versionStateFactory({ data: "2.10.0" }),
+    }),
+    machine: machineStateFactory({
+      active: "abc123",
+      items: [],
+    }),
   });
+});
+
+afterEach(() => {
+  jest.useRealTimers();
+});
+
+it("can show if a machine is currently commissioning", () => {
+  state.machine.items = [
+    machineDetailsFactory({
+      fqdn: "test.maas",
+      status: NodeStatus.COMMISSIONING,
+      system_id: "abc123",
+    }),
+  ];
+
+  renderWithMockStore(<StatusBar />, { state });
+
+  expect(screen.getByTestId("status-bar-status")).toHaveTextContent(
+    "test.maas: Commissioning in progress..."
+  );
+});
+
+it("can show if a machine has not been commissioned yet", () => {
+  state.machine.items = [
+    machineDetailsFactory({
+      commissioning_start_time: "",
+      fqdn: "test.maas",
+      system_id: "abc123",
+    }),
+  ];
+
+  renderWithMockStore(<StatusBar />, { state });
```

Comment:
It's a common practice to use a custom render function/wrapper to reduce boilerplate[1]. Would be good to get your thoughts on the approach.

I'm aware I'm setting a precedent here and I'm happy to revert this if needed,  but keep in mind that I plan to get back to this soon and make sure we're consistent in however we decide to approach this[2].

[1] - https://testing-library.com/docs/react-testing-library/setup/#custom-render
[2] - [Reduce boilerplate in testing-library tests ](https://github.com/canonical-web-and-design/app-tribe/issues/791)

Reply:
I usually try not be dogmatic about DRY approaches but this seems sensible to me. Might as well export `WithMockStoreProvider` too, for cases where we might want to use `rerender`.

Reply:
Yeah that seems like a nice way of reducing the duplication while still allowing us the flexibility of setting up the Redux state as we need it.

Reply:
ok great thanks @Caleb-Ellis and @huwshimi 🙏 

> I usually try not be dogmatic about DRY approaches

I'm with you on this!

> Might as well export `WithMockStoreProvider` too, for cases where we might want to use `rerender`.

Oh that's a good point, didn't think about `rerender`. I think we can use a custom `rerender` function within `renderWithMockStore` instead, should be cleaner. I'll look into this.